client/components/data/purchases/edit-card-details/index.jsx

```diff
@@ -55,24 +54,27 @@ const EditCardDetailsData = React.createClass( {
 	mixins: [ observe( 'sites' ) ],
 
 	componentWillMount() {
-		fetchStoredCards();
 		if ( shouldFetchUserPurchases( PurchasesStore.get() ) ) {
 			fetchUserPurchases( user.get().ID );
 		}
 	},
 
 	render() {
 		return (
-			<StoreConnection
-				cardId={ this.props.cardId }
-				component={ this.props.component }
-				getStateFromStores={ getStateFromStores }
-				hasLoadedSites={ this.props.sites.fetched }
-				isDataLoading={ isDataLoading }
-				loadingPlaceholder={ this.props.loadingPlaceholder }
-				purchaseId={ this.props.purchaseId }
-				selectedSite={ this.props.sites.getSelectedSite() }
-				stores={ stores } />
+			<div>
+				<QueryStoredCards />
```

Comment:
Now that we're using `connect` to provide access to the global store through props, we don't need data components. Once this and #6647 are merged, we can remove this component entirely, so we should move any changes to it (like adding `QueryStoredCards`) into `EditCardDetails`.